### ISSUE TITLE
Inverse control when image flipped

### ISSUE THREAD

**Issue by benjR** [EliasKotlyar/Xiaomi-Dafang-Hacks#257]
Hi there!

Been using firmware, works quite good so far (I have to test the detection stuff though) !

Anyway, when you use the RSTP functionnality, the controls are not inversed, which means that when you ask the camera to go to the right, it goes to the left, same goes for up that goes down, it's not very easy to use that way...

Thanks

**Comment by freddujour** [comment]
Hi,
If you need tester, I can support. My dafang is also turned upside down and this wrong behaviour is turning the track mode completely unusable.

**Comment by xehn** [comment]
Okay, I've got this working on my camera here.  Not sure if this is the "right" way to fix it, but it works for me.

In /system/sdcard/bin, I renamed the binary file **motor** to **motor.bin**.  
Then I created a new file **motor** with the following contents:

```
#!/bin/sh

# Check for orientation setting 
if [ `/system/sdcard/bin/setconf -g f` -eq 1 ]; then
    # We are flipped, so check direction variable and reverse
    case $2 in
         u) flip=d ;;
         d) flip=u ;;
         r) flip=l ;;
         l) flip=r ;;
         *)  ;;
    esac
else
    flip=$2
fi    

# Send command to motor binary
/system/sdcard/bin/motor.bin $1 $flip $3 $4
```
Basically, it intercepts calls to the motor binary.  It checks to see if the camera is flipped.  If it is, the script will flip/mirror the direction argument, and then pass it on to the binary.

It is more of a hack than a proper fix, but it works.

**Comment by nik0** [comment]
Did you test with the (maybe not used) detection tracking feature ?

**Comment by xehn** [comment]
I did some brief testing (consisting of dancing around in front of the camera).  It seemed to track accurately.

**Comment by staudtm** [comment]
When will this function be implemented?

**Comment by xehn** [comment]
@staudtm - if you need this functionality, there are instructions for how to implement several posts above yours.

**Comment by codedmind** [comment]
How to make this change persistent after upgrade?
More, how to persistent the flip state after reboot/upgrade

**Comment by jmtatsch** [comment]
@codedmind flip persistence is now implemented in #569

**Comment by codedmind** [comment]
@jmtatsch the persistence is solved, but in the settings page the icon (on/off) isn't updated.
Other issue that still exists is that the directions aren't inverted when the image is fliped

**Comment by jmtatsch** [comment]
@codedmind I know, that is why I didn't close the issue yet. Maybe you can help a little with the pointed out issues?

**Comment by codedmind** [comment]
@jmtatsch the issues for now is "solved" with @xehn solution, but only works until next update

**Comment by nik0** [comment]
The PR is coming ...

**Comment by nik0** [comment]
PR created, thanks @xehn , I used your code

**Comment by jmtatsch** [comment]
Fixed in #570
